manifests/init.pp

```diff
@@ -231,6 +231,7 @@
   $facts_to_tags = [],
   $puppet_run_reports = false,
   $puppetmaster_user = $settings::user,
+  $puppet_gem_provider = $datadog_agent::params::gem_provider,
```

Comment:
Let's make this:
```
String $puppet_gem_provider = $datadog_agent::params::gem_provider,
```

Feel free to move this further down if it looks awkward having surrounding arguments without a specific data type. It's unfortunate we have this situation where some have data types specified while other parameters do not :( We can probably clean this up in a separate PR when we have the bandwidth.

Reply:
done. Yes we might want to clean this up